### PR TITLE
Implement Merge Sort and Generic Search with custom compare function

### PR DESCRIPTION
**Pull request by TravorLZH**
I clean the history by re-forking your repository. (#12)

### CODE REVIEW

**Comment by ParadoxZero** [comment]
Great work! 

Please refactor it to comply with the code guide.

**Comment by TravorLZH** [comment]
> Please refactor it to comply with the code guide.

What's the code guide

**Comment by ParadoxZero** [comment]
Code guide: https://github.com/ParadoxZero/libgeneric/wiki/Code-guide

**Comment by TravorLZH** [comment]
> If a global constant, suffix it with `g`.

Please change `suffix` to `prefix`

File: include/algorithm/mergesort.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+/** @brief Merge Sort for integers
+ *
+ * @param arr:	Array being sorted
+ * @param n:	Length of the array
+ *
+ * @return	Pointer to the sorted array
+ *		Always equal to the input on success
+ *		NULL on failure
+ */
+extern int *mergeSort(int *arr,size_t n);
```
Comment (ParadoxZero) [inline include/algorithm/mergesort.h:52]: This function is not needed for the API

**Review by ParadoxZero — APPROVED** [review]
Some changes needed. But they are small and I feel guilty about sending so many change request 😝 

So I will do them myself tomorrow xD

**Pull request merged**